pkg/apis/vsphereprovider/v1alpha1/vsphereproviderconfig_types.go

```diff
@@ -52,8 +52,38 @@ type VSphereMachineProviderSpec struct {
 	// machine is cloned.
 	// +optional
 	DiskGiB int32 `json:"diskGiB,omitempty"`
+	// Snapshot is the name of the snapshot from which the VM was cloned
+	// +optional
+	Snapshot string `json:"snapshot"`
+
+	// CloneMode specifies the type of clone operation.
+	// The LinkedClone mode is only support for templates that have at least
+	// one snapshot. If the template has no snapshots, then CloneMode defaults
+	// to FullClone.
+	// When LinkedClone mode is enabled the DiskGiB field is ignored as it is
+	// not possible to expand disks of linked clones.
+	// Defaults to LinkedClone, but fails gracefully to FullClone if the source
+	// of the clone operation has no snapshots.
+	// +optional
+	CloneMode CloneMode `json:"cloneMode,omitempty"`
 }
 
+// CloneMode is the type of clone operation used to clone a VM from a template.
+type CloneMode string
+
+const (
+	// FullClone indicates a VM will have no relationship to the source of the
+	// clone operation once the operation is complete. This is the safest clone
+	// mode, but it is also the fastest.
```

Comment:
This description and the one below claim to be the fastest clones 😅  I suspect this one is actually not the fastest? Can you double check and update the comments appropriately

Reply:
Haha, this is from upstream https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/419f258418b152034bf4eed039cc947e80ad02a4/api/v1alpha3/types.go#L39

Reply:
Probably best to just leave it then 😅 Didn't realise that was a copy/pasta from upstream

Reply:
https://kubernetes.slack.com/archives/CKFGK3SSD/p1584524099203800